src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,51 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Objects;
+
 public class SalaryInfo {
+    private static final int DATE = 0;
+    private static final int NAME = 1;
+    private static final int HOURS = 2;
+    private static final int INCOME = 3;
+    private static final int DAY = 1;
+    private static final String DATE_FORMAT = "dd.MM.yyyy";
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate dateFromDateFormat = changeDateFormat(dateFrom);
+        LocalDate dateToDateFormat = changeDateFormat(dateTo);
+        int[] salary = new int[names.length];
+
+        for (String dataRow : data) {
+            String[] dataInfo = dataRow.split(" ");
+            if (changeDateFormat(dataInfo[DATE]).isAfter(dateFromDateFormat.minusDays(DAY))
+                    && changeDateFormat(dataInfo[DATE]).isBefore(dateToDateFormat.plusDays(DAY))) {
```

Comment:
The date range check should ensure inclusivity by using `!isBefore(dateFromDateFormat)` and `!isAfter(dateToDateFormat)` instead of adjusting the dates with `minusDays` and `plusDays`.